compiler/rustc_expand/src/mbe/macro_rules.rs

```diff
@@ -647,6 +647,7 @@ fn check_lhs_no_empty_seq(sess: &ParseSess, tts: &[mbe::TokenTree]) -> bool {
                 if seq.separator.is_none()
                     && seq.tts.iter().all(|seq_tt| match seq_tt {
                         TokenTree::MetaVarDecl(_, _, Some(NonterminalKind::Vis)) => true,
+                        TokenTree::Token(t) => matches!(t, Token { kind: DocComment(..), .. }),
```

Comment:
if you slightly restructure the code you should be able to track whether a doc comment was used. If it was, you should attach an extra note to the diagnostic saying `doc comments are ignored in matcher position`, because otherwise it's quite confusing. You may even be able to record the span of the last doc comment inside and emit a `span_note` pointing at it for maximal prettyness 😊

Reply:
The same could be done for a `:vis` matcher as it's not 100% obvious there either that they can be empty.

Also there might be cases where you have multiple such matchers combined:

```Rust
macro_rules! foo {
($(
    ///
    $v:vis
)*) => {};
}
```

It is a nice thing for a separate PR as well, shrug.

Reply:
doing it together with vis would be a good follow up PR, agreed. r=me too if you want :D